pkg/replicate/scheme.go

```diff
@@ -49,11 +51,16 @@ func NewBlockFilter(
 	for _, compactionLevel := range compactionLevels {
 		allowedCompactions[compactionLevel] = struct{}{}
 	}
+	blockSet := make(map[string]struct{})
```

Comment:
The blockID list will be super short. Can we just loop over it instead of using map? Might be simpler (:

Reply:
Sounds better!